### PR TITLE
defer integration tests

### PR DESCRIPTION
**Pull request by Geal**
This adds support for testing `@defer` in integration tests

---

**Checklist**

Complete the checklist (and note appropriate exceptions) before the PR is marked ready-for-review.

- [ ] Changes are compatible[^1]
- [ ] Documentation[^2] completed
- [ ] Performance impact assessed and acceptable
- Tests added and passing[^3]
    - [ ] Unit Tests
    - [ ] Integration Tests
    - [ ] Manual Tests

**Exceptions**

*Note any exceptions here*

**Notes**

[^1]: It may be appropriate to bring upcoming changes to the attention of other (impacted) groups. Please endeavour to do this before seeking PR approval. The mechanism for doing this will vary considerably, so use your judgement as to how and when to do this.
[^2]: Configuration is an important part of many changes. Where applicable please try to document configuration examples.
[^3]: Tick whichever testing boxes are applicable. If you are adding Manual Tests, please document the manual testing (extensively) in the Exceptions.

### CODE REVIEW

File: apollo-router/tests/common.rs [hunk 1]
```diff
@@ -582,7 +582,6 @@ impl IntegrationTest {
 
                 let mut request = builder.json(&query).build().unwrap();
                 telemetry.inject_context(&mut request);
-                request.headers_mut().remove(ACCEPT);
```
Comment (Geal) [inline apollo-router/tests/common.rs:585]: @BrynCooke do you remember why the integration tests were removing the `Accept` header from the request?

**Comment by Geal** [comment]
interestingly, multer is unable to parse the multipart body we send with defer and subs:
```
--graphql
content-type: application/json

{"data":null,"errors":[{"message":"HTTP fetch failed from 'accounts': 404: Not Found","path":[],"extensions":{"code":"SUBREQUEST_HTTP_ERROR","service":"accounts","reason":"404: Not Found","http":{"status":404}}},{"message":"HTTP fetch failed from 'accounts': subgraph response does not contain 'content-type' header; expected content-type: application/json or content-type: application/graphql-response+json","path":[],"extensions":{"code":"SUBREQUEST_HTTP_ERROR","service":"accounts","reason":"subgraph response does not contain 'content-type' header; expected content-type: application/json or content-type: application/graphql-response+json","http":{"status":404}}}],"hasNext":true}
--graphql
content-type: application/json

{"hasNext":false}
--graphql--
```

it fails with the error `incomplete multipart stream`. Maybe some different assumptions in how we chunk the response?

**Comment by Geal** [comment]
the multer failure was due to a boundary reported as a quoted string

**Comment by Geal** [comment]
Supporting subscriptions in those tests would be challenging, as they are not designed to execute tasks asynchronously, so we cannot(for now) use the HTTP callback while we are already waiting for a response. We could have subgraph emulation support websockets though

**Review by bnjjj — APPROVED** [review]

**Pull request merged**